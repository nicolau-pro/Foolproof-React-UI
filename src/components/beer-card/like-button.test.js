import React from 'react';
import ReactDOM from 'react-dom';
import Like from './like-button';

describe('LikeButton', () => {
  it('should be defined', () => {
    expect(Like).toBeDefined();
  });

  it('renders correctly when liked is not defined', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Like />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly when liked=true', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Like liked={true} />, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly when liked=false', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Like liked={false} />, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});

// A lot more tests could have been performed by simulating click actions on the Like button
