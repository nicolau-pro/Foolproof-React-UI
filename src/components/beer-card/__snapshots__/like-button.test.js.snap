// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton renders correctly when liked=false 1`] = `
<div>
  <button
    class="sc-bdVaJa gvsZWK"
  >
    <svg
      class="sc-bwzfXH jrkcCm"
    >
      <path
        d="M16,28.261c0,0-14-7.926-14-17.046c0-9.356,13.159-10.399,14-0.454c1.011-9.938,14-8.903,14,0.454
 C30,20.335,16,28.261,16,28.261z"
        id="heart-icon"
      />
    </svg>
  </button>
</div>
`;

exports[`LikeButton renders correctly when liked=true 1`] = `
<div>
  <button
    class="sc-bdVaJa gvsZWK"
  >
    <svg
      class="sc-bwzfXH jvdhts"
    >
      <path
        d="M16,28.261c0,0-14-7.926-14-17.046c0-9.356,13.159-10.399,14-0.454c1.011-9.938,14-8.903,14,0.454
 C30,20.335,16,28.261,16,28.261z"
        id="heart-icon"
      />
    </svg>
  </button>
</div>
`;
